src/test/java/org/sonatype/gradle/plugins/scan/ossindex/TransportBuilderTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2020-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonatype.gradle.plugins.scan.ossindex;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.sonatype.ossindex.service.client.transport.UserAgentSupplier;
+
+import org.gradle.api.Project;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.gradle.util.GradleVersion;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TransportBuilderTest
+{
+  private TransportBuilder builder;
+
+  @Before
+  public void setup() {
+    builder = new TransportBuilder();
+  }
+
+  @Test
+  public void testBuild() {
+    Project project = ProjectBuilder.builder().build();
+    assertThat(builder.build(project)).isNotNull();
+  }
+
+  @Test
+  public void testBuildUserAgentSupplier() throws IOException {
```

Comment:
Minor: I'm probably just missing it, but I couldn't see where the file: 
client.properties -> userAgent=Sonatype_Nexus_Gradle
gets used. Wondering if it might be worth a simple test to make sure userAgent is set as expected. (Or not, I just couldn't quite see where it happens).

Reply:
> Minor: I'm probably just missing it, but I couldn't see where the file:
> client.properties -> userAgent=Sonatype_Nexus_Gradle
> gets used. Wondering if it might be worth a simple test to make sure userAgent is set as expected. (Or not, I just couldn't quite see where it happens).

The file was added previously for the IQ Server integration: https://github.com/sonatype-nexus-community/scan-gradle-plugin/blob/master/src/main/resources/com/sonatype/insight/client.properties

Both user agent and version are currently being used internally by the IQ Server client. For OSS Index I'm just using that existing file to take the version from it for the custom user agent to send to OSS Index.